### PR TITLE
Suggestions for TCP-TLS docs

### PR DESCRIPTION
**Pull request by schavis**
Suggestions for https://github.com/hashicorp/vault/pull/27318

### CODE REVIEW

File: website/content/docs/configuration/listener/tcp/index.mdx [hunk 1]
```diff
@@ -47,6 +47,42 @@ also omit keys from the response when the corresponding value is empty (`""`).
...
+  vulnerability scanners may flag issues with your configuration.
+
+  In particular, Sweet32 (CVE-2016-2183) is an attack against 64-bit block size
+  ciphers including 3DES that may allow an attacker to break the encryption of
+  long lived connections. According to vulnerability disclosures, Sweet32 took a
+  single HTTPS session with 785 GB of traffic to break the encryption.
+
+  As of May 2024, the Go team does not believe the risk of Sweet32 is sufficient
+  to remove existing client compatibility by deprecating 3DES support, however,
+  the team did de-prioritize .
```
Comment (finnigja) [inline website/content/docs/configuration/listener/tcp/index.mdx:78]:
```suggestion
  the team did [de-prioritize 3DES](https://github.com/golang/go/issues/45430) in favor of AES-based ciphers.
```

File: website/content/docs/configuration/listener/tcp/index.mdx [hunk 2]
```diff
@@ -47,6 +47,42 @@ also omit keys from the response when the corresponding value is empty (`""`).
...
+
+<Note title="Sweet32 and 3DES">
+
+  The Go team and HashiCorp believe that the set of cyphers supported by `tls`
+  and `tlsutil` is appropriate for modern, secure usage. However, some
+  vulnerability scanners may flag issues with your configuration.
+
+  In particular, Sweet32 (CVE-2016-2183) is an attack against 64-bit block size
+  ciphers including 3DES that may allow an attacker to break the encryption of
+  long lived connections. According to vulnerability disclosures, Sweet32 took a
```
Comment (finnigja) [inline website/content/docs/configuration/listener/tcp/index.mdx:73]:
```suggestion
  long lived connections. According to the [vulnerability disclosure](https://sweet32.info/), Sweet32 took a
```

File: website/content/docs/configuration/listener/tcp/tcp-tls.mdx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+<CodeBlockConfig hideClipboard highlight="5-7">
+
+```plaintext
+listener "tcp" {
+  address = "127.0.0.1:8200"
+  tls_cert_file = "cert.pem"
+  tls_key_file  = "key.pem"
+  tls_min_version = "tls12"
+  tls_max_version = "tls12"
+  tls_cipher_suites = "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256"}
```
Comment (finnigja) [inline website/content/docs/configuration/listener/tcp/tcp-tls.mdx:64]:
```suggestion
  tls_cipher_suites = "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256"
}
```

**Review by jonathanfrappier — APPROVED** [review]

File: website/content/docs/configuration/listener/tcp/index.mdx (hunk 2 above)
Comment (finnigja) [reply website/content/docs/configuration/listener/tcp/index.mdx:73]: Bringing in relevant link from draft (context may be useful for reader).

File: website/content/docs/configuration/listener/tcp/index.mdx (hunk 1 above)
Comment (finnigja) [reply website/content/docs/configuration/listener/tcp/index.mdx:78]: Bringing in relevant link from draft (context may be useful for reader).

File: website/content/docs/configuration/listener/tcp/tcp-tls.mdx (hunk 3 above)
Comment (finnigja) [reply website/content/docs/configuration/listener/tcp/tcp-tls.mdx:64]: Bringing in other ciphersuites from draft, as this is more likely to be a useful example that just listing one ciphersuite.

**Review by finnigja — COMMENTED** [review]
A few small suggestions but generally looks good to me!

**Comment by schavis** [comment]
Replaced with https://github.com/hashicorp/vault/pull/27335

**Pull request closed without merging**